commands/connect/diagnose.js

```diff
@@ -50,18 +56,32 @@ module.exports = {
     context.region = connection.region_url
     let results = yield cli.action('Diagnosing connection', co(function * () {
       let url = '/api/v3/connections/' + connection.id + '/validations'
-      return yield api.request(context, 'GET', url)
+      try {
+        let {data: {result_url: resultUrl}} = yield api.request(context, 'POST', url)
+
+        while (true) {
```

Comment:
Maybe we should bail after a number of attempts? I think if something is wrong and the resultURL keeps responding with a 404 or something this will run forever?

Reply:
fixed.